src/compile/axis/rules.ts

```diff
@@ -78,11 +79,10 @@ export function orient(channel: Channel) {
 
 export function tickCount(channel: Channel, fieldDef: FieldDef<string>, scaleType: ScaleType, size: VgSignalRef) {
 
-  if (!fieldDef.bin && !hasDiscreteDomain(scaleType) && scaleType !== 'log') {
-    // Vega's default tickCount often lead to a lot of label occlusion on X without 90 degree rotation
-    // Thus, we set it to 5 for width = 200
-    // and set the same value for y for consistency.
-
+  if (!hasDiscreteDomain(scaleType) && scaleType !== 'log') {
+    if (fieldDef.bin) {
+      return {signal: `min(ceil(${size.signal}/40), ${(fieldDef.bin as Bin).maxbins})`};
```

Comment:
Why removing comments?  Instead, please add comments saying why it should be min of the two?